internal/ingress/controller/template/template.go

```diff
@@ -595,14 +595,16 @@ func buildUpstreamName(host string, b interface{}, loc interface{}) string {
 
 	upstreamName := location.Backend
 
-	for _, backend := range backends {
-		if backend.Name == location.Backend {
-			if backend.SessionAffinity.AffinityType == "cookie" &&
-				isSticky(host, location, backend.SessionAffinity.CookieSessionAffinity.Locations) {
-				upstreamName = fmt.Sprintf("sticky-%v", upstreamName)
+	if !dynamicConfigurationEnabled{
```

Comment:
@zrdaley Why are we only enabling cookie based session affinity if endpoint awareness is disabled?

Reply:
One more thought on this - session affinity being ignored if endpoint awareness is enabled is very implicit; if you wind up PR-ing this upstream, you might have people wondering why their cookie based session affinity is broken when they have endpoint awareness enabled as well.

Would suggest maybe a warning that session affinity is being ignored if dynamic configuration is enabled? Alternately, if you wanted to be stricter - you could throw an error. I'm not 100% on what is best practice here, so I'll defer to you :)